app/lib/services/backend/firestore/firestore_backend_service.dart

```diff
@@ -453,14 +461,99 @@ class FirestoreBackendService extends BackendServiceAggregator {
-              authorId: message['userId'] as String,
-            );
-          }).toList();
+          messages
+              .mapIndexed(
+                (int index, Map<String, dynamic> message) =>
+                    ChatBackendServiceMessage(
+                  content: message['content'] as String? ?? '',
+                  authorId: message['authorId'] as String? ?? '',
+                  messageId: index.toString(),
+                  timestamp: message['timestamp'] != null
+                      ? DateTime.parse(message['timestamp'] as String)
+                      : DateTime.now(),
+                ),
+              )
+              .toList();
+
+  @override
+  Future<void> addChatMessage(
+    String chatId,
+    ChatBackendServiceMessage message,
+  ) async {
+    final DocumentReference<Map<String, dynamic>> chatReference =
+        firestore.collection(_chatsCollection).doc(chatId);
+
+    final DocumentSnapshot<Map<String, dynamic>> chatSnapshot =
+        await chatReference.get();
+
+    if (!chatSnapshot.exists) {
+      throw Exception("Chat data not found");
+    }
+
+    final Map<String, dynamic>? chatData = chatSnapshot.data();
+
+    if (chatData == null) {
+      throw Exception("Chat data not found");
+    }
+
+    final List<dynamic> messages =
+        (chatData['messages'] as List<dynamic>?) ?? <List<dynamic>>[];
+
+    final Map<String, dynamic> newMessage = <String, dynamic>{
+      'content': message.content,
+      'authorId': message.authorId,
+      'timestamp': message.timestamp.toIso8601String(),
+    };
+
+    messages.add(newMessage);
+
+    return chatReference.update(<String, dynamic>{
+      'messages': messages,
+    });
+  }
+
+  @override
+  Future<void> deleteChatMessage(String chatId, String messageId) async {
+    final DocumentReference<Map<String, dynamic>> chatReference =
+        firestore.collection(_chatsCollection).doc(chatId);
+
+    final DocumentSnapshot<Map<String, dynamic>> chatSnapshot =
+        await chatReference.get();
+
+    if (!chatSnapshot.exists) {
+      throw Exception("Chat data not found");
+    }
+
+    final Map<String, dynamic>? chatData = chatSnapshot.data();
+
+    if (chatData == null) {
+      throw Exception("Chat data not found");
+    }
+
+    final List<dynamic> messages =
+        (chatData['messages'] as List<dynamic>?) ?? <List<dynamic>>[];
+
+    messages.removeAt(int.parse(messageId));
+
+    return chatReference.update(<String, dynamic>{
+      'messages': messages,
+    });
+  }
+
+  @override
+  Future<ChatBackendServicePerson> getCurrentUser() async {
+    final QuerySnapshot<Map<String, dynamic>> value =
+        await firestore.collection(_usersCollection).get();
+    final QueryDocumentSnapshot<Map<String, dynamic>> user = value.docs.first;
+
+    final String? imageUrl = user['imageUrl'] as String?;
+
+    return ChatBackendServicePerson(
+      id: user.id,
+      name: user['name'] as String,
+      imageUrl: imageUrl ?? '',
```

Comment:
**suggestion:** I would recommend just making the field Optional since the image url is not a important data field. If its missing we can just show a placeholder. The Widget we created for this already implements this logic so lets use it :)